numpy/lib/_iotools.py

```diff
@@ -50,6 +54,13 @@ def _is_bytes_like(obj):
     return True
 
 
+def _is_path(obj):
+    """
+    Check whether obj is a pathlib.Path object.
+    """
+    return PurePath is not None and isinstance(obj, PurePath)
```

Comment:
> `PurePath` looks like the wrong type to check as it may be purely abstract, `Path` would seem better.

Ah, I misread the docs and thought `Path` was an alias for `PosixPath` or `WindowsPath`. You're right, those both inherit from `Path`.

However, it doesn't look to me like `PurePath` is ever abstract, and I'm not sure it matters anyway: I'm only using it for `isinstance`, so its fine if its abstract.

That said, there's no reason to not just go with `Path`, so maybe we should do that.